app/Library/Notifications/Drivers/NativeDriver.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * Pony.fm - A community for pony fan music.
+ * Copyright (C) 2016 Josef Citrine
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Poniverse\Ponyfm\Library\Notifications\Drivers;
+
+
+use Config;
+use Poniverse\Ponyfm\Contracts\Favouritable;
+use Poniverse\Ponyfm\Models\Activity;
+use Poniverse\Ponyfm\Models\Comment;
+use Poniverse\Ponyfm\Models\Playlist;
+use Poniverse\Ponyfm\Models\Track;
+use Poniverse\Ponyfm\Models\User;
+use Minishlink\WebPush\WebPush;
+
+class NativeDriver extends AbstractDriver {    
+    /**
+     * Method for sending notifications to devices
+     *
+     * @param Activity $activity
+     * @param User[] $recipients collection of {@link User} objects
+     */
+    private function pushNotifications(Activity $activity, $recipients)
+    {
+        if (Config::get('ponyfm.gcm_key') != 'default') {
```

Comment:
Loose comparisons should not be used. Always use `===` in new code, along with explicit casts if necessary.

As well, it looks like this is designed to fail silently if no key is set. `null` should be used to signify the lack of a key, and at least a log entry should be generated if a key was expected and not found.